### PR TITLE
New Content Ignite adapter

### PR DESCRIPTION
**Pull request by jdrucey**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

Adding a new adapter for Content Ignite
Initially based on AdButler as used by our backend, but soon to introduce video and native

- test parameters for validating bids
```
{
  bidder: 'contentignite',
  params: {
    accountID: '168237',
    zoneID: '299680'
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer: jamie@contentignite.com
- [x] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo: https://github.com/prebid/prebid.github.io/pull/670

### CODE REVIEW

**Comment by jdrucey** [comment]
@bretg Reverted those hello world changes! Cheers.

File: modules/contentigniteBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    return serverRequests
+  },
+
+  interpretResponse: function(serverResponse, bidRequest) {
+    var bidObj = bidRequest.bidRequest
+    var bidResponses = []
+    var bidResponse = {}
+    var isCorrectSize = false
+    var isCorrectCPM = true
+    var CPM
```
Comment (jsnellbaker) [inline modules/contentigniteBidAdapter.js]: Does `CPM` need to be in all caps?  Can you make it consistent with the other variables syntax (camelCase)?

File: modules/contentigniteBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+      // Ensure that response ad matches one of the placement sizes.
+      utils._each(bidObj.sizes, function(size) {
+        if (width === size[0] && height === size[1]) {
+          isCorrectSize = true
+        }
+      })
+      if (isCorrectCPM && isCorrectSize) {
+        bidResponse.requestId = bidObj.bidId
+        bidResponse.bidderCode = spec.code
```
Comment (jsnellbaker) [inline modules/contentigniteBidAdapter.js]: The `bidderCode` does not need to be populated here; it's actually managed by the prebid code in our `bidderFactory`'s code.  Can you please remove this line?

Comment (jdrucey) [reply modules/contentigniteBidAdapter.js]: This has now been removed

File: modules/contentigniteBidAdapter.js (hunk 1 above)
Comment (jdrucey) [reply modules/contentigniteBidAdapter.js]: Have reverted to lowercase `cpm`

File: modules/contentigniteBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    var height
+
+    serverResponse = serverResponse.body
+    if (serverResponse && serverResponse.status === 'SUCCESS' && bidObj) {
+      CPM = serverResponse.cpm
+      minCPM = utils.getBidIdParameter('minCPM', bidObj.params)
+      maxCPM = utils.getBidIdParameter('maxCPM', bidObj.params)
+      width = parseInt(serverResponse.width)
+      height = parseInt(serverResponse.height)
+      console.log(serverResponse.cpm)
```
[the remark this answers is not in the document]
Comment (jdrucey) [reply modules/contentigniteBidAdapter.js]: Sorry about that! i could have sworn i removed it! ha

File: modules/contentigniteBidAdapter.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
+'use strict'
```
[the remark this answers is not in the document]
Comment (jdrucey) [reply modules/contentigniteBidAdapter.js]: Ive now introduced arrow functions, const/let usage and reformatted with semi-colons. Would be really neat if there could be a Prettier config file added to the Prebid.js project. It is something we use a lot our end to auto-format js, and while it is very opinionated (by design) there are some settings for project based preference, like the semi-colons in this case. Not sure if this has been considered before, but just an idea!

**Comment by jdrucey** [comment]
Hi @jsnellbaker, thanks for taking the time to look over the code, we appreciate it. 
All your suggestions are very sensible, and i agree it is good to bring this code up-to-date with regards to ES6 standards, i certainly prefer it myself.
I have pushed the fixes and changes so let me know if all is as expected.

**Pull request merged**